blockservice/blockservice.go

```diff
@@ -67,7 +46,7 @@ func (s *BlockService) AddBlock(b *blocks.Block) (key.Key, error) {
 	if err != nil {
 		return k, err
 	}
-	if err := s.worker.HasBlock(b); err != nil {
+	if err := s.Exchange.HasBlock(context.TODO(), b); err != nil {
```

Comment:
thinking that this HasBlock is a passive activity, not an active one like the "AddBlock" call. i.e. user is adding, not wanting to write out. or send blocks to others. may be worth doing .HasBlock in a goroutine/process/worker.

context should be derived from some context passed to the blockservice initially, not the current request, i think.